### PR TITLE
Saving models

### PR DESCRIPTION
**Pull request by jrwalsh1**
Adds some basic functionality to save models as HDFStore objects and represent layers and models as metadata configurations.

The models currently consist of layers and weights.  The layer parameters and the weight matrix are saved as DataFrames in the h5 file, and a configuration for the model (mostly metadata for the layers) is also saved as an attribute.

This PR is ready, but there is more to do:
  - reading models from saved files
  - including state parameters from fitting into a saved model (should they go into the model itself?), e.g. optimizers
  - tests

### CODE REVIEW

**Comment by drckf** [comment]
As we are saving in HDF5 format, each class could just have its own key `model`, `optimizers`, etc.

File: paysage/models/hidden.py [hunk 1]
```diff
@@ -34,17 +37,56 @@ def __init__(self, layer_list):
...
+            None
+
+        Returns:
+            A dictionary configuration for the model.
+
+        """
+        config = {
+        "model type": "RBM",
+        "layers": [ly.get_config() for ly in self.layers],
+        "layer_types": ["visible", "hidden"],
```
Comment (drckf) [inline paysage/models/hidden.py]: Why is `layer_types` needed? The zeroth layer is always visible and all others are always hidden, so it should be contained in the layer order of the list

Comment (jrwalsh1) [inline paysage/models/hidden.py]: This is a placeholder for now for layer names, which we'll need for deep models. It probably should go into the layer as an attribute, though, now or at that point.

**Comment by jrwalsh1** [comment]
For saving more than just the model data, the question for me is who is doing the saving. The fit method is currently the only object that knows about everything. We could always have a 'save' utils function, or a state object.

File: paysage/models/hidden.py (hunk 1 above)
Comment (drckf) [inline paysage/models/hidden.py]: Why are layer names needed for deep models?

**Comment by drckf** [comment]
If the `model` (or other classes) save function took an open `pandas.HDFStore` object it could just write it's own data. Multiple objects can simply write to the same open stream.

**Pull request merged**